src/native/fs/fs_napi.cpp

```diff
@@ -883,6 +937,18 @@ struct Writefile : public FSWorker
         } else if ((size_t)len != _len) {
             SetError(XSTR() << "Writefile: partial write error " << DVAL(len) << DVAL(_len));
         }
+
+        if (_set_xattr) {
+            if (_xattr_need_fsync) {
+                SYSCALL_OR_RETURN(fsync(fd));
```

Comment:
Again, trying to understand - we fsync before setting xattr intentionally?

Reply:
That makes sense. But what tier2 FS does when XFS fsync fails (data block write failure)? Does tier2 filesystem fsync fails? If yes, do we not mark it for migration?

Reply:
On a second thought. Are we even supposed to handle fsync failures? If no, then my comment is useless.

Reply:
fsync can fail with EIO or ENOSPC. 
the question is what else can we do on these cases.
now we fail the block write, and the endpoint will retry - see [retry_write_block()](https://github.com/noobaa/noobaa-core/blob/dcf47166771b814524e44cafab2895ea2a5d3b7f/src/sdk/map_client.js#L251)
so for now I think this is a good starting point to fail and try again, up to some retry limit.
this is also related to the question we had on writing with O_DIRECT...
We should revise this if needed in the future.
Thanks. Resolving